internal/lsp/server.go

```diff
@@ -307,73 +269,60 @@ func (l *LanguageServer) StartConfigWorker(ctx context.Context) {
 }
 
 func (l *LanguageServer) StartCommandWorker(ctx context.Context) {
+	// note, in this function conn.Call is used as the workspace/applyEdit message is a request, not a notification
+	// as per the spec. In order to be 'routed' to the correct handler on the client it must have an ID
+	// receive responses too.
+	// Note however that the responses from the client are not needed by the server.
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case params := <-l.commandRequest:
-			switch params.Command {
-			case "regal.fmt":
-				edits, target, err := l.formatToEdits(params, format.Opts{})
-				if err != nil {
-					l.logError(err)
+			var editParams *ApplyWorkspaceEditParams
 
-					break
-				}
+			var err error
 
-				editParams := ApplyWorkspaceEditParams{
-					Label: "Format using opa fmt",
-					Edit: WorkspaceEdit{
-						DocumentChanges: []TextDocumentEdit{
-							{
-								TextDocument: OptionalVersionedTextDocumentIdentifier{URI: target},
-								Edits:        edits,
-							},
-						},
-					},
-				}
+			var fixed bool
 
-				// note, here conn.Call is used as the workspace/applyEdit message is a request, not a notification
-				// as per the spec. In order to be 'routed' to the correct handler on the client it must have an ID
-				// receive responses too.
-				err = l.conn.Call(
-					ctx,
-					methodWorkspaceApplyEdit,
-					editParams,
-					nil, // however, the response content is not important
+			switch params.Command {
+			case "regal.fmt":
+				fixed, editParams, err = l.fixEditParams(
+					"Format using opa fmt",
+					&fixes.Fmt{OPAFmtOpts: format.Opts{}},
+					commands.ParseOptions{TargetArgIndex: 0},
+					params,
 				)
-				if err != nil {
-					l.logError(fmt.Errorf("failed %s notify: %v", methodWorkspaceApplyEdit, err.Error()))
-				}
 			case "regal.fmt.v1":
-				edits, target, err := l.formatToEdits(params, format.Opts{RegoVersion: ast.RegoV0CompatV1})
-				if err != nil {
-					l.logError(err)
+				fixed, editParams, err = l.fixEditParams(
+					"Format for Rego v1 using opa-fmt",
+					&fixes.Fmt{OPAFmtOpts: format.Opts{RegoVersion: ast.RegoV0CompatV1}},
+					commands.ParseOptions{TargetArgIndex: 0},
+					params,
+				)
+			case "regal.use-assignment-operator":
```

Comment:
For consistency, perhaps we should rename `regal.fmt` -> `regal.opa-fmt` and `regal.fmt.v1` -> `regal.use-rego-v1` above?

Reply:
Or perhaps `regal.fix.opa-fmt` and so on, in case we want to add commands later that aren't about editing.

Reply:
Yeah, I think that makes sense.

Reply:
Fixed in 3a1d447bd0ef6ab6305e73408880565b445d9501